### PR TITLE
Backport customlogs to OSS

### LINKED ISSUE #2090: Introduce Custom Logs apis
### Description

Introduce the BE components for supporting Custom Logs

### Acceptance Criteria

All Custom Logs-related code is in OSS.

### PR DESCRIPTION
**Pull request by alxarch**
## Background

Closes #2090
Customlogs goes OSS!

## Changes

- Backports `preprocessors` `logschema`, `customlogs` modules
- Backports `logtypesapi` custom log stuff
- Wires up logtype resolvers with logtypes API for log processor and datacatalog updater
- Update permissions
- Backports `customlogs` CLI tool
- Decouples `logtypes` from `parsers` module

## Testing

- mage test:go
- mage test:cfn
- deployed and ran for ~2hours with no errors in the log processor or the datacatalog updater

### CODE REVIEW

**Comment by kostaspap** [comment]
Does this close #2090 ?  If so please add a `Closes #2090` comment in your description so that we know the ticket is resolved once this PR merges.

File: internal/core/logtypesapi/api.go [hunk 1]
```diff
@@ -35,10 +38,71 @@ const LambdaName = "panther-logtypes-api"
...
+const (
+	// ErrRevisionConflict is the error code to use when there is a revision conflict
+	ErrRevisionConflict = "RevisionConflict"
+	ErrAlreadyExists    = "AlreadyExists"
+	ErrNotFound         = "NotFound"
+	ErrInUse            = "InUse"
+)
+
+// APIError is an error that has a code and a message and is returned as part of the API response
+type APIError struct {
```
Comment (giorgosp) [inline internal/core/logtypesapi/api.go:70]: Nice, we need to have such error responses in the other API's too.

**Review by giorgosp — APPROVED** [review]

File: internal/core/logtypesapi/api.go (hunk 1 above)
Comment (austinbyers) [reply internal/core/logtypesapi/api.go:70]: Yep, improved (and standardized) error handling is one of the goals of the API consolidation project

File: deployments/log_analysis.yml [hunk 2]
```diff
@@ -731,10 +732,9 @@ Resources:
           Statement:
             - Effect: Allow
               Action: lambda:InvokeFunction
-              Resource: !Sub arn:${AWS::Partition}:lambda:${AWS::Region}:${AWS::AccountId}:function:panther-logtypes-api
-            - Effect: Allow
-              Action: s3:List*
-              Resource: !Sub arn:${AWS::Partition}:s3:::${ProcessedDataBucket}*
```
Comment (austinbyers) [inline deployments/log_analysis.yml:737]: Did you mean to remove this permission? I think the sync command still needs to list the processed data bucket?

File: deployments/bootstrap_gateway.yml [hunk 3]
```diff
@@ -227,7 +227,6 @@ Resources:
                 - !Sub arn:${AWS::Partition}:lambda:${AWS::Region}:${AWS::AccountId}:function:panther-organization-api
                 - !Sub arn:${AWS::Partition}:lambda:${AWS::Region}:${AWS::AccountId}:function:panther-source-api
                 - !Sub arn:${AWS::Partition}:lambda:${AWS::Region}:${AWS::AccountId}:function:panther-users-api
-                - !Sub arn:${AWS::Partition}:lambda:${AWS::Region}:${AWS::AccountId}:function:panther-datacatalog-updater
```
Comment (austinbyers) [inline deployments/bootstrap_gateway.yml:230]: Why was this removed? I don't see any associated changes to the custom resources Lambda function

**Review by austinbyers — COMMENTED** [review]
Not sure about some CFN changes

File: deployments/log_analysis.yml (hunk 2 above)
Comment (alxarch) [reply deployments/log_analysis.yml:737]: These were leftovers from the datacatalog refractor. Sync does not have to list the S3 bucket for a long time (ever since gluesync fix).

File: deployments/bootstrap_gateway.yml (hunk 3 above)
Comment (alxarch) [reply deployments/bootstrap_gateway.yml:230]: All communication with datacatalog goes through sqs. The permission was a leftover.
